libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/IComponentMemoryScopes.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Bot.Builder.Dialogs.Memory.Scopes;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Dialogs.Declarative
+{
+    public interface IComponentMemoryScopes
```

Comment:
>IComponentMemoryScopes [](start = 21, length = 22)

Did you consider deriving from a base class?  We've been down the interface->class route a couple of times.

Reply:
Yeah, I didn't because there isn't one library which defines assets that are consumable.  They are distributed.  And there is only 1 method, so this is essentially a well known delegate.

---
In reply to: [376541449](https://github.com/microsoft/botbuilder-dotnet/pull/3350#discussion_r376541449) [](ancestors = 376541449)